EXILED/Exiled.API/Extensions/MirrorExtensions.cs

```diff
@@ -167,20 +170,51 @@ public static ReadOnlyDictionary<string, string> RpcFullNames
+        [Obsolete("This method is not working. Use PlayGunSound(Player, Vector3, ItemType, float, int, bool) overload instead.")]
         public static void PlayGunSound(this Player player, Vector3 position, ItemType itemType, byte volume, byte audioClipId = 0)
         {
-            // TODO: Not finish
-            /*
-            GunAudioMessage message = new()
+        }
+
+        /// <summary>
+        /// Plays a gun sound that only the <paramref name="player"/> can hear.
+        /// </summary>
+        /// <param name="player">Target to play.</param>
+        /// <param name="position">Position to play on.</param>
+        /// <param name="itemType">Weapon' sound to play.</param>
+        /// <param name="pitch">Speed of sound.</param>
+        /// <param name="clipIndex">Index of clip.</param>
+        public static void PlayGunSound(this Player player, Vector3 position, ItemType itemType, float pitch = 1, int clipIndex = 0)
+        {
+            if (!itemType.IsWeapon(false) || itemType == ItemType.ParticleDisruptor)
+                return;
+
+            using (NetworkWriterPooled writer = NetworkWriterPool.Get())
             {
-                Weapon = itemType,
-                AudioClipId = audioClipId,
-                MaxDistance = volume,
-                ShooterHub = player.ReferenceHub,
-                ShooterPosition = new RelativePosition(position),
-            };
+                writer.WriteUShort(NetworkMessageId<RoleSyncInfo>.Id);
+                new RoleSyncInfo(Server.Host.ReferenceHub, RoleTypeId.ClassD, player.ReferenceHub).Write(writer);
+                writer.WriteRelativePosition(new RelativePosition(0, 0, 0, 0, false));
+                writer.WriteUShort(0);
+                player.Connection.Send(writer);
+            }
+
+            Features.Items.Firearm firearm = Item.Get<Features.Items.Firearm>(Server.Host.Inventory.ServerAddItem(itemType, ItemAddReason.AdminCommand));
+            firearm.BarrelAmmo = 1;
+            firearm.BarrelMagazine.IsCocked = true;
+            player.SendFakeSyncVar(Server.Host.Inventory.netIdentity, typeof(Inventory), nameof(Inventory.NetworkCurItem), firearm.Identifier);
 
-            player.Connection.Send(message);*/
+            if (!firearm.Base.TryGetModule(out AudioModule audioModule))
+                return;
+
+            Timing.CallDelayed(0.1f, () => // Seems like without this delay firearm is not 'ready'
+            {
+                audioModule.SendRpc(player.ReferenceHub, writer =>
+                    audioModule.ServerSend(writer, clipIndex, pitch, MixerChannel.Weapons, 12f, position, false));
+
+                player.SendFakeSyncVar(Server.Host.Inventory.netIdentity, typeof(Inventory), nameof(Inventory.NetworkCurItem), ItemIdentifier.None);
+                firearm.Destroy();
+
+                player.Connection.Send(new RoleSyncInfo(Server.Host.ReferenceHub, RoleTypeId.None, player.ReferenceHub));
```

Comment:
This need to set Host to it's real role not a specific one in case a plugin use this for something else it's could crash it